src/uct/base/uct_mem.c

```diff
@@ -215,6 +218,9 @@ ucs_status_t uct_mem_alloc(size_t length, const uct_alloc_method_t *methods,
                 break;
             }
 
+            alloc_length = length;
+            address      = (params->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_ADDRESS) ?
+                           params->address : NULL;
```

Comment:
we an just use "length" for this (since posix_memalign does not change it)

Reply:
but `alloc_length` could be changed by uct_md_mem_alloc call but md_mem_alloc failed, right? unless failure in md_mem_alloc is a fatal error and no other methods are attempted.

Reply:
It does turn out to be a non-recoverable error. I'll remove this.